connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorManager.java

```diff
@@ -190,6 +193,33 @@ private void activateConnector(InboundConnectorDefinitionImpl newConnector) {
     }
   }
 
+  private void handlePollingConnector(
+      PollingConnectorExecutable executable,
+      InboundConnectorDefinitionImpl newConnector,
+      InboundConnectorContextImpl inboundContext) {
+    Function<Long, Map<String, String>> fetchVariablesFunction =
+        processDefinitionSearch::fetchVariablesByProcessInstanceKey;
+    Supplier<List<ProcessInstance>> processInstancesSupplier =
+        createProcessInstancesSupplier(fetchVariablesFunction, newConnector.processDefinitionKey());
+    ProcessInstanceContext processInstanceContext =
+        new ProcessInstanceContextImpl(processInstancesSupplier, new FeelEngineWrapper());
+    InboundIntermediateConnectorContextImpl intermediateContext =
+        new InboundIntermediateConnectorContextImpl(inboundContext, processInstanceContext);
+
+    executable.poll(intermediateContext);
+  }
+
+  private Supplier<List<ProcessInstance>> createProcessInstancesSupplier(
+      Function<Long, Map<String, String>> fetchVariablesFunction, Long processDefinitionKey) {
+    return () ->
+        processDefinitionSearch.fetchProcessInstancesByDefinitionKey(processDefinitionKey).stream()
+            .map(
+                instance ->
+                    new ProcessInstance(
+                        instance.getKey(), fetchVariablesFunction.apply(instance.getKey())))
+            .collect(Collectors.toList());
+  }
```

Comment:
This should live in the implementation of the `InboundIntermediateConnectorContextImpl` itself.